tests/integration_tests/validation/validation_util.py

```diff
@@ -70,11 +93,48 @@ def get_test_data(args, expected_failure_ids=frozenset(), expected_empty_testcas
         nonunique_test_ids = {test_id: count for test_id, count in test_id_frequencies.items() if count > 1}
         if nonunique_test_ids:
             raise Exception(f'Some test IDs are not unique.  Frequencies of nonunique test IDs: {nonunique_test_ids}.')
-        nonexistent_expected_failure_ids = expected_failure_ids - set(test_id_frequencies)
-        if nonexistent_expected_failure_ids:
-            raise Exception(f"Some expected failure IDs don't match any test cases: {sorted(nonexistent_expected_failure_ids)}.")
+        if expected_failure_ids:
+            nonexistent_expected_failure_ids = expected_failure_ids - set(test_id_frequencies)
+            if nonexistent_expected_failure_ids:
+                raise Exception(f"Some expected failure IDs don't match any test cases: {sorted(nonexistent_expected_failure_ids)}.")
         return results
     finally:
         cntlr.modelManager.close()
         PackageManager.close()
         PluginManager.close()
+
+
+original_normalize_url_function = WebCache.normalizeUrl
+
+
+def normalize_url_function(config: ConformanceSuiteConfig):
+    def normalize_url(self, url, base=None):
+        if config.url_replace:
+            if url.startswith(config.url_replace):
+                return url.replace(config.url_replace, f'{config.local_filepath}/')
+        return original_normalize_url_function(self, url, base)
+    return normalize_url
+
+
+def get_conformance_suite_test_results(config: ConformanceSuiteConfig, log_to_file: bool = False, offline: bool = False):
+    args = [
+        '--file', os.path.abspath(os.path.join(config.local_filepath, config.file)),
```

Comment:
> Also, it seems like this is going to run against the ZIP instead of the extract directory for the suites that need extraction.
The only suite that uses `extract_path` currently is IXBRL, and it only uses the extracted path for the `--packages` value. The unextracted path works fine passed in as the main `--file` value. If we do end up encountering a conformance suite where the entry point needs to be extracted outside of the zip, the `extract_path` functionality will need to be slightly reworked, but it works now.

Reply:
It passed in the previous implementation because CONFORMANCE_SUITE was the path to the extracted directory, and both `--file` and `--packages` navigated to that directory.

In my implementation, `--file` is still referencing the .zip, but that works because it doesn't try to access the nested .zip through that route. `--packages` navigates to the extracted directory, which is the only reason the .zip is extracted to begin with: arelle can't read nested zips. I'm guessing you used a pre-existing inlineXBRL-1.1-conformanceSuite-2020-04-08.zip when you ran locally? That would fail for the same reason it's failing on my fork, because the version we've used thus far (the version that's on S3) has an extra directory level that the .zip downloaded from xbrl.org doesn't. Same reason XBRL 2.1 is failing.

Once this PR is ready for QA, I can swap out the XBRL 2.1 and IXBRL zips on S3 and re-run those, and I expect them to pass.